### PR TITLE
Too many getPointIdentity, #1574

### LINKED ISSUE #1574: Regression, new bug, related to useMsgLikelihoods
Looks like it has something to do with `getSolverParams(fg).useMsgLikelihoods = true`.  MWE:

```julia
using IncrementalInference

fg = generateGraph_CaesarRing1D()
getSolverParams(fg).useMsgLikelihoods = true

solveGraph!(fg)
```

Code completes fine if `useMsgLikelihoods=false`

_Originally posted by @dehann in https://github.com/JuliaRobotics/IncrementalInference.jl/issues/1572#issuecomment-1194719417_

### PR DESCRIPTION
**Pull request by dehann**
close:
- #1574

### CODE REVIEW

**Comment by dehann** [comment]
The best way to avoid this forcing solution in the future is to fix
- JuliaRobotics/ApproxManifoldProducts.jl#41

first, and make sure that the manifold tangent vectors coming from the MKD sampling are consistent with Manifolds.jl all cases.

**Comment by dehann** [comment]
made a few more changes, trying to get all these samplePoint and sampleTangent functions to use `getPointIdentity`

**Comment by Affie** [comment]
> made a few more changes, trying to get all these samplePoint and sampleTangent functions to use getPointIdentity

They shouldn’t, it written to support manifolds that are not groups in the future.

**Comment by dehann** [comment]
just trying to get the test to work again.  Think that is what may have broke the tests.  If you have a minute during the day, could you perhaps just look this over to see if you can quickly identify the issue here please?  I commited with I have sofar.

Might be stuck on:
- JuliaManifolds/Manifolds.jl#512

**Comment by dehann** [comment]
oh, test are failing on Sphere2 vs Sphere(2)

**Comment by Affie** [comment]
>  If you have a minute during the day, could you perhaps just look this over to see if you can quickly identify the issue here please?

Ok, I'll take over this branch then.

**Comment by Affie** [comment]
> Coordinates should always be a vector

My first instinct was correct, it is a change in Manifolds.jl that showed this bug of the coordinates as a matrix.

File: src/ManifoldSampling.jl [hunk 1]
```diff
@@ -49,12 +48,12 @@ function samplePoint(M::AbstractDecoratorManifold, sbelief, p=getPointIdentity(M
...
 end
 
-function samplePoint(M::AbstractDecoratorManifold, sbelief::ManifoldKernelDensity, p=identity_element(M, mean(sbelief)), retraction_method::AbstractRetractionMethod=ExponentialRetraction())
+function samplePoint(M::AbstractDecoratorManifold, sbelief::ManifoldKernelDensity, p=getPointIdentity(M), retraction_method::AbstractRetractionMethod=ExponentialRetraction())
   X = sampleTangent(M, sbelief, p)
   return retract(M, p, X, retraction_method)
 end
 
-function samplePoint(x::ManifoldKernelDensity, p=mean(x)) 
+function samplePoint(x::ManifoldKernelDensity, p=getPointIdentity(x.manifold))
```
Comment (dehann) [inline src/ManifoldSampling.jl:56]:
```suggestion
function samplePoint(x::ManifoldKernelDensity, p=mean(x)) 
```

File: src/Factors/GenericFunctions.jl [hunk 2]
```diff
@@ -31,7 +31,7 @@ end
 
 #::MeasurementOnTangent
 function distanceTangent2Point(M::SemidirectProductGroup, X, p, q)
-  q̂ = Manifolds.compose(M, p, exp(M, identity_element(M, p), X)) #for groups
+  q̂ = Manifolds.compose(M, p, exp(M, getPointIdentity(M), X)) #for groups
```
[the remark this answers is not in the document]
Comment (dehann) [reply src/Factors/GenericFunctions.jl:34]: okay, just this one to change back?

File: src/services/ApproxConv.jl [hunk 3]
```diff
@@ -22,12 +22,15 @@ function approxConvBelief(dfg::AbstractDFG,
   partl = collect(1:len)[ mask ]
 
   # is the convolution infoPerCoord full or partial
+  vari = getVariable(dfg, target)
+  M = getManifold(vari)
+  vT = getVariableType(vari)
   if sum(mask) == len
     # not partial
-    return manikde!(getManifold(getVariable(dfg, target)), pts, partial=nothing)
+    return manikde!(M, pts, partial=nothing, u0=getPointIdentity(vT))
```
Comment (Affie) [inline src/services/ApproxConv.jl:30]: This errored so left it out, but didn't look at why it didn't work.

File: src/Factors/GenericFunctions.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (dehann) [reply src/Factors/GenericFunctions.jl:34]:
```suggestion
  q̂ = Manifolds.compose(M, p, exp(M, identity_element(M, p), X)) #for groups
```

File: test/testSphereMani.jl [hunk 4]
```diff
@@ -9,8 +9,9 @@ using Test
 @testset "Test Sphere(2) prior and relative" begin
 ##
 
+# NOTE us getPointIdentity instead
 #FIXME REMOVE! this is type piracy and not a good idea, for testing only!!!
-Manifolds.identity_element(::Sphere{2, ℝ}, p::Vector{Float64}) = Float64[1,0,0]
+# Manifolds.identity_element(::Sphere{2, ℝ}, p::Vector{Float64}) = Float64[1,0,0]
```
Comment (Affie) [inline test/testSphereMani.jl:14]: I believe this is needed for AMP

**Comment by Affie** [comment]
@dehann, what is your reason for changing the `identity_element(M, p)` calls to `getPointIdentity(M)`?

**Pull request closed without merging**